### ISSUE TITLE
Cannot load P2 binaries

### ISSUE THREAD

**Issue by GitRayman** [parallaxinc/Propeller-Tool#22]
Does not appear to be able to load P2 binary files.

**Comment by PropGit** [comment]
Right, the v2.0.0.1 Alpha didn't have any binary file handling updated to support the P2 yet.  There's a note on the releases page that intended to mean this (and more), but I think I'll clarify it.

[**WARNING**] Object Info not supported for P2 (currently). While the edit's Object Vew works, the View Info function is not usable for P2. _**Object Info window's Load/Save button functionality should also be avoided for now.**_

**Comment by GitRayman** [comment]
Doesn't look like I can save as binary either...  Thought I could compare this binary with the one produced by PNut, but doesn't look like I can...

**Comment by PropGit** [comment]
We can logic-capture what's sent to the P2 via each just to verify that it's different, as it seems it must be.

**Comment by PropGit** [comment]
I found differences in the data sent (PNut vs. Propeller Tool).  Don't know why yet.  The vast majority of the first 93K of data transmitted is identical; however, it does vary in two clusters starting around 105 bytes.  After 93,247 bytes, the data the Propeller Tool sends is very different from PNut, and then it's truncated around 105 Kbytes in the transmission.

I'll definitely have to explore this further to determine the cause and fix it.

**Comment by PropGit** [comment]
Looks like the FILE directive handling is bad.

If I comment out the ```file "TestPat1.bmp" '640 x 480, 8bpp bitmap``` line and download with PNut and Propeller Tool, they both deliver 17,739 bytes total to the P2 (all of which are identical).

**Comment by GitRayman** [comment]
Ok, yes looks like file is bad…

 

I replaced with just some reserved space and it works.

Ray

 


 

 

 




Raymond Allen, Ph.D.

Plasma Physics Division, Code 6770

T 202.404.5324    F 202.767.2012

www.nrl.navy.mil

**Comment by PropGit** [comment]
Confirmed there's a problem with the Propeller Tool's FILE directive handling.

If I simplify the code to be only:
~~~
DAT
orgh        
BitmapFile 
                file "1000Bytes.dat"
                byte 0[1000]
~~~
where 1000Bytes.dat is a file with exact 1000 bytes, Propeller Tool and PNut transmit the same data to the Propeller.

However, if I make the code use TestPat1.bmp:
~~~
DAT
orgh        
BitmapFile 
                file "TestPat1.bmp" '640 x 480, 8bpp bitmap
                byte 0[1000]
~~~
The output (of the TestPat1.bmp part of the stream) appears to be truncated after 87,418 bytes, then it's followed by the encoded 0's.

**Comment by PropGit** [comment]
> Ok, yes looks like file is bad… I replaced with just some reserved space and it works. 

I can't say that the file is bad.  If that were the case we should see PNut have the same output.  I still think Propeller Tool is tripping over the file somewhere in the stream.

**Comment by PropGit** [comment]
Issue #21 relates to this thread.

**Comment by GitRayman** [comment]
Sorry, I meant the "file" directive is bad...

**Comment by GitRayman** [comment]
I'm sure the bitmap file itself is good.
